pkg/kubelet/util/pluginwatcher/plugin_watcher.go

```diff
@@ -232,6 +280,10 @@ func (w *Watcher) Stop() error {
 	}()
 	select {
 	case <-c:
+		// forget registered plugins
+		for key, _ := range w.registeredPlugins {
```

Comment:
although test call Watcher.stop() directly, but in real case, it only happen when kubelet restart, then
plugins need to register again, so it would make sense to clean up. 

but if there is the case that Watcher need be stop and start, but plugins stay registered.
i am happy to understand the case.

Reply:
although test call Watcher.stop() directly, but in real case, it only happen when kubelet restart, then
plugins need to register again, so it would make sense to clean up.

but if there is the case that Watcher need be stop and start, but plugins stay registered.
i am happy to understand the case.

Reply:
In the restart case your map will be recreated through a call to `NewWatcher`.
I don't understand when you need this map to be cleaned up and re-used?

Reply:
you are right, in the restart case, it don't need to call Watcher.stop().

i think the confusion comes from who is going to call watcher.stop()? 
is it only for testing and simulating (simulate kubelet restart -> watcher restart)?
if yes, then clean-up is part of that fresh re-start state simulation.

@vikaschoudhary16  could you share your thought here?

Reply:
I'm not talking about watcher.Stop here, I'm talking about the next line:
```golang
+			w.registeredPlugins[key] = make(map[string]bool)
```

Reply:
@RenaudWasTaken yes, if you look at the original test, it 
  ```    
     registerPlugin()
     watcher.Stop()
     watcher.Start()
     register the same plugin again, and expecting register success
```

so i don't want to break this behavior, it means it will try to clean up when watcher.Stop(), so that the same plugin can register success after watcher.Start().

i have updated the change to address your other comments, please take a look, thanks

Reply:
Ah I see why you are doing this now.
Restarting the watcher as you described is not really the expected pattern and introduces this weird dependency to cleanup the watcher map.

In that case it would make more sense to not re-use the watcher and recreate it.
Do you mind re-writing it as follows:
i.e:
```golang

watcher := newWatcher(...)
registerPlugin()
watcher.Stop()

watcher = newWatcher(...)
watcher.Start()
```

Thanks!